prov/verbs/src/ep_rdm/verbs_rdm_cm.c

```diff
@@ -378,37 +443,56 @@ fi_ibv_rdm_tagged_process_connect_request(struct rdma_cm_event *event,
 		assert (conn->id[idx] == NULL);
 		conn->id[idx] = id;
 
-		fi_ibv_rdm_tagged_prepare_conn_memory(ep, conn);
+		ret = fi_ibv_rdm_tagged_prepare_conn_memory(ep, conn);
+		if (ret != FI_SUCCESS)
+			goto err;
+
 		fi_ibv_rdm_tagged_init_qp_attributes(&qp_attr, ep);
-		rdma_create_qp(id, ep->domain->pd, &qp_attr);
+		if (rdma_create_qp(id, ep->domain->pd, &qp_attr)) {
+			ret = -errno;
+			goto err;
+		}
 		conn->qp[idx] = id->qp;
+		/* TODO: check err handling after merge */
 		if (ep->rq_wr_depth !=
 		    fi_ibv_rdm_tagged_repost_receives(conn, ep,
 						      ep->rq_wr_depth))
 		{
 			VERBS_INFO(FI_LOG_AV, "repost receives failed\n");
-			return -FI_ENOMEM;
+			ret = -FI_ENOMEM;
+			goto err;
 		}
 
 		id->context = conn;
 
 		fi_ibv_rdm_pack_cm_params(&cm_params, conn, ep);
 
-		ret = rdma_accept(id, &cm_params);
-		assert(ret == 0);
+		if (rdma_accept(id, &cm_params)) {
+			VERBS_INFO_ERRNO(FI_LOG_AV, "rdma_accept\n", errno);
+			ret = -errno;
+			goto err;
+		}
 		if (cm_params.private_data) {
 			free((void *) cm_params.private_data);
 		}
-	} 
+	}
+out:
 	return ret;
+err:
+	if (conn) {
+		/* ret err code is already set here, just cleanup resources */
+		fi_ibv_rdm_tagged_conn_cleanup(conn);
+		free(conn);
+	}
+	goto out;
```

Comment:
No, in addition free is already calling inside fi_ibv_rdm_tagged_conn_cleanup. I'm fixing it.
Thanks!